src/permissions.rs

```diff
@@ -0,0 +1,63 @@
+use users::{ Groups, Users };
+use users::os::unix::GroupExt;
+
+pub fn check_group_membership<U: Users+ Groups>(cache: &U, expected_group_names: &[&str]) {
+    let mut is_one_group_found = false;
+    let mut i = 0;
+    let expected_group_size = expected_group_names.len();
+
+    while !is_one_group_found && i < expected_group_size {
+        let group_name = expected_group_names[i];
+
+        if let Some(group) = cache.get_group_by_name(group_name) {
+            is_one_group_found = true;
+            let current_uid = cache.get_current_uid();
+            let user = cache.get_user_by_uid(current_uid).unwrap();
+
+            if !group.members().contains(&user.name().to_owned()) {
+                panic!("Not a member of the {} group", group_name);
+            }
+        }
```

Comment:
Thanks for the background. I agree with you this would have been a major issue if we wanted to keep the PR.